sherpa/data.py

```diff
@@ -464,17 +487,13 @@ class Data(NoNewAttributesAfterInit, BaseData):
 
     def __init__(self, name, indep, y, staterror=None, syserror=None):
         """
-        Warning: Currently, Data objects ignore any masks on input data. If you have masked data,
-        set ``data.mask`` after initialization - and note that the Sherpa convention for masking
-        is OPPOSITE to numpy, i.e. in Sherpa True marks a valid value and False an invalid,
-        to-be-ignored value.
-
         Parameters
         ----------
         name : basestring
             name of this dataset
         indep: tuple of array_like
-            the tuple of independent arrays
+            the tuple of independent arrays. If this is an numpy masked array,
+            the mask will used.
```

Comment:
what does 'the mask will be used' mean? Is it "the masked array will be used" or "the unmasked values" or ...? I don't use the masked array stuff so I don't know the expected terminology.

Actually, this is the independent array: isn't the mask ignored in this case? Actually, it depends on _init_data_space does, which isn't immediately obvious.